### PR TITLE
[quant] Add QuantFusion to graph_executor

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26591 [quant] Add QuantFusion to graph_executor**

Summary:
att

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D17636651](https://our.internmc.facebook.com/intern/diff/D17636651)

### CODE REVIEW

**Comment by jerryzh168** [comment]
btw this PR is regenerated, last PR is merged somehow.

File: torch/csrc/jit/graph_executor.cpp [hunk 1]
```diff
@@ -708,6 +711,11 @@ void runNondiffOptimization(std::shared_ptr<Graph>& graph) {
   // Rewrite subgraphs with many MMs into expressions that batch them.
   BatchMM(graph);
 
+  if (at::globalContext().qEngine() == at::kFBGEMM ||
+      at::globalContext().qEngine() == at::kQNNPACK) {
```
Comment (ZolotukhinM) [inline torch/csrc/jit/graph_executor.cpp]: Optimization pipeline in graph executor cannot be conditionalized on a global state. We need to make sure QuantFusion is always correct and run it unconditionally.

**Review by ZolotukhinM — CHANGES_REQUESTED** [review]

File: torch/csrc/jit/graph_executor.cpp [hunk 2]
```diff
@@ -1,5 +1,7 @@
 #include <torch/csrc/jit/graph_executor.h>
 
+#include <aten/src/ATen/Context.h>
+
```
Comment (ZolotukhinM) [inline torch/csrc/jit/graph_executor.cpp]: This is unused.

**Review by ZolotukhinM — APPROVED** [review]
Looks good!

**Pull request closed without merging**